src/components/ActionMenu/components/Actions/Actions.tsx

```diff
@@ -35,32 +56,92 @@ export function Actions({actions = [], groups = []}: Props) {
     () => setActiveMenuGroup(undefined),
     [],
   );
-  const menuActions = [...actions, ...groups];
 
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const handleResize = useCallback(
+    debounce(
+      () => {
+        if (!newDesignLanguage || !actionsLayoutRef.current) return;
+        setAvailableWidth(actionsLayoutRef.current.offsetWidth);
+      },
+      20,
+      {leading: false, trailing: true, maxWait: 40},
+    ),
+    [actionsLayoutRef],
+  );
+
+  useEffect(() => {
+    if (!actionsLayoutRef.current) return;
+    setAvailableWidth(actionsLayoutRef.current.offsetWidth);
+  }, [actionsLayoutRef]);
+
+  useEffect(() => {
+    if (!menuGroupRef.current) return;
+    menuGroupWidthRef.current = menuGroupRef.current?.offsetWidth;
+  }, [menuGroupRef]);
```

Comment:
A `ref` never changes, its content (`current`) changes, so you don't need to have it explicit.